### PR TITLE
[JENKINS-71715] Update minimum supported Jenkins version to 2.440 to be able to use new `ConsoleUrlProvider` API to link to Pipeline build console page

### PR DESCRIPTION
**Pull request by dwnusbaum**
See [JENKINS-71715](https://issues.jenkins.io/browse/JENKINS-71715) and https://github.com/jenkinsci/jenkins/pull/8321. 

### Testing done

Tested in combination with https://github.com/jenkinsci/pipeline-input-step-plugin/pull/145 and https://github.com/jenkinsci/pipeline-graph-view-plugin/pull/289, see this video:

https://github.com/jenkinsci/pipeline-input-step-plugin/assets/1068968/de3788fb-c41f-48ba-9420-561e82e2a354

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -63,7 +63,7 @@
     </pluginRepositories>
     <properties>
         <changelist>999999-SNAPSHOT</changelist>
-        <jenkins.version>2.361.4</jenkins.version>
+        <jenkins.version>2.418-rc34070.e1b_938b_3d266</jenkins.version> <!-- TODO: https://github.com/jenkinsci/jenkins/pull/8321 -->
```
Comment (dwnusbaum) [inline pom.xml:66]:
```suggestion
        <jenkins.version>2.420-rc34121.c34df9dd8347</jenkins.version> <!-- TODO: https://github.com/jenkinsci/jenkins/pull/8321 -->
```

File: pom.xml [hunk 2]
```diff
@@ -63,7 +63,7 @@
     </pluginRepositories>
     <properties>
         <changelist>999999-SNAPSHOT</changelist>
-        <jenkins.version>2.414</jenkins.version>
+        <jenkins.version>2.420-rc34121.c34df9dd8347</jenkins.version> <!-- TODO: https://github.com/jenkinsci/jenkins/pull/8321 -->
```
Comment (timja) [inline pom.xml:66]:
```suggestion
        <jenkins.version>2.420</jenkins.version>
```

**Comment by timja** [comment]
ready to go @dwnusbaum?

**Comment by dwnusbaum** [comment]
@timja Yeah I was just waiting for the new LTS to drop; I'll try to update all of the related PRs this week.

**Review by jglick — APPROVED** [review]
Sanity tested recently?

**Pull request merged**